modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearOptimisticTxPrepareFuture.java

```diff
@@ -155,30 +154,8 @@ private void onError(Throwable e, boolean discoThread) {
             }
         }
 
-        if (ERR_UPD.compareAndSet(this, null, e)) {
-            boolean marked = tx.setRollbackOnly();
-
-            if (e instanceof IgniteTxRollbackCheckedException) {
-                if (marked) {
-                    tx.rollbackAsync().listen(new IgniteInClosure<IgniteInternalFuture<IgniteInternalTx>>() {
-                        @Override public void apply(IgniteInternalFuture<IgniteInternalTx> fut) {
-                            try {
-                                fut.get();
-                            }
-                            catch (IgniteCheckedException e) {
-                                U.error(log, "Failed to automatically rollback transaction: " + tx, e);
-                            }
-
-                            onComplete();
-                        }
-                    });
-
-                    return;
-                }
-            }
-
+        if (ERR_UPD.compareAndSet(this, null, e))
```

Comment:
Could you please briefly describe why the following calls are no longer relevant? I mean `tx.setRollbackOnly()` and `tx.rollbackAsync()`.

Reply:
Removal of this block fixes the inability to rollback optimistic transaction if failed during preparing. New behavior and reproducing test is introduced by GG-22420. Otherwise deadlock happens while trying to rollback preparing transaction on local node. Removal of this block doesn't seem to break anything visible to TC tests.